### ISSUE TITLE
Timeout

### ISSUE THREAD

**Issue by dgutson** [justanhduc/task-spooler#46]
I suggest to add the ability to set a timeout for running tasks, killing them on expiration.

**Comment by justanhduc** [comment]
Hi @dgutson. Yes it will be a good feature to add. I will come up with an implementation soon.

**Comment by dgutson** [comment]
Thanks, sometimes tasks hang occupying a slot this turning the whole process slower, not only CPU usage.
If I set `task -S 4` but 3 hang overnight, I end up having no parallelism and a night or weekend is lost.
This is a real use case.

**Comment by justanhduc** [comment]
Yeah I myself did need this feature just a few weeks ago. I needed to restart `ffmpeg` recording regularly to prevent it from freezing, but at that time I had to make do with some hack in bash script.
